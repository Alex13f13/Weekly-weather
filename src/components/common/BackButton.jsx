import KeyboardBackspaceIcon from "@mui/icons-material/KeyboardBackspace";
import IconButton from "@mui/material/IconButton";
import { useNavigate } from "react-router-dom";

export default function BackButton() {
	const navigate = useNavigate();
	const handleGoBack = () => {
		navigate(-1);
	};

	return (
		<IconButton onClick={handleGoBack} aria-label="back" size="large">
			<KeyboardBackspaceIcon />
		</IconButton>
	);
}
