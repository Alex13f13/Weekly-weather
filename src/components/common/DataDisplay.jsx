import { createElement } from "react";
import { HTML_TAGS } from "../../utils/constants";

export default function DataDisplay({ data, label = "", units = "", tag = HTML_TAGS.p }) {
	const selectedTag = HTML_TAGS[tag] || HTML_TAGS.p;

	return data && createElement(selectedTag, null, `${label} ${data} ${units}`);
}
