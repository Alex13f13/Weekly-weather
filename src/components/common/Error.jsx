import styled from "styled-components";

const ErrorNoDataContainer = styled.div`
	width: 90%;
	height: auto;
	background-color: #f8d7da;
	color: #721c24;
	padding: 10px;
	border: 1px solid #f5c6cb;
	border-radius: 4px;
	margin-top: 20px;
`;

const ErrorMessage = styled.p`
	margin: 0;
	text-align: center;
`;

export default function ErrorNoData() {
	return (
		<ErrorNoDataContainer>
			<ErrorMessage>Hubo un error al cargar los datos.</ErrorMessage>
		</ErrorNoDataContainer>
	);
}
