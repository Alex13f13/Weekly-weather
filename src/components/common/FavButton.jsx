import IconButton from "@mui/material/IconButton";
import FavoriteIcon from "@mui/icons-material/Favorite";
import HeartBrokenIcon from "@mui/icons-material/HeartBroken";
import { SIZE, LABEL } from "../../utils/constants";

export default function FavButton({ isFavorite, onClick = () => {} }) {
	return (
		<IconButton aria-label={LABEL.favourite} size={SIZE.large} onClick={onClick}>
			{isFavorite ? <HeartBrokenIcon /> : <FavoriteIcon />}
		</IconButton>
	);
}
