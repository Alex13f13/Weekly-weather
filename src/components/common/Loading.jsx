import { LABEL } from "../../utils/constants";

export default function Loading() {
	return <>{LABEL.loading}</>;
}
