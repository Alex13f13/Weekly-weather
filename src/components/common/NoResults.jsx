export default function NoResults() {
	return <p>No hay resultados</p>;
}
