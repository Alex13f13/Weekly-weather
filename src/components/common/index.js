export { default as BackButton } from './BackButton';
export { default as DataDisplay } from './DataDisplay';
export { default as Error } from './Error';
export { default as FavButton } from './FavButton';
export { default as Loading } from './Loading';
export { default as NoResults } from './NoResults';
export { default as Selector } from './Selector';
export { default as StatusManager } from './StatusManager';
export { default as WeatherCard } from './WeatherCard';
export { default as WeatherCardList } from './WeatherCardList';
