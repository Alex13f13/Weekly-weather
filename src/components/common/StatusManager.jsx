import Loading from "./Loading";
import Error from "./Error";
import NoResults from "./NoResults";

export default function StatusManager({ isLoading, error, noResults, children }) {
	if (isLoading) return <Loading />;
	if (error) return <Error />;
	if (noResults) return <NoResults />;

	return <>{children}</>;
}
