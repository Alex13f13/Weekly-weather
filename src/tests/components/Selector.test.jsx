import { render, screen, fireEvent } from "@testing-library/react";
import { Selector } from "../../components/common";

describe("Selector Component", () => {
	const label = "Select Option";
	const options = [
		{ id: "option1", name: "Option 1" },
		{ id: "option2", name: "Option 2" },
		{ id: "option3", name: "Option 3" },
	];
	const defaultValue = "option2";

	it("should render the label and options correctly", () => {
		render(<Selector label={label} options={options} defaultValue={defaultValue} />);

		const selectLabel = screen.getByText(label);
		expect(selectLabel).toBeDefined();

		options.forEach((option) => {
			const selectOption = screen.getByText(option.name);
			expect(selectOption).toBeDefined();
		});

		const select = screen.getByRole("combobox");
		const selectedValue = select.value;

		expect(selectedValue).toBe(defaultValue);
	});

	it("should call the onSelect function when an option is selected", () => {
		let onSelectCalled = "";
		const onSelectMock = (selectedOption) => {
			onSelectCalled = selectedOption;
		};

		render(<Selector label={label} options={options} onSelect={onSelectMock} />);

		const select = screen.getByRole("combobox");

		fireEvent.change(select, { target: { value: "option2" } });

		expect(onSelectCalled).toBe("option2");
	});

	it("should match snapshot", () => {
		const { container } = render(
			<Selector label={label} options={options} defaultValue={defaultValue} />
		);

		expect(container).toMatchSnapshot();
	});
});
