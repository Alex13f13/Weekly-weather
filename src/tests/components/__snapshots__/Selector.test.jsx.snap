// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Selector Component > matches snapshot 1`] = `
<div>
  <div
    class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-standard MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-standard css-1c2i806-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="true"
    >
      Select Option
    </label>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl MuiNativeSelect-root css-v4u5dn-MuiInputBase-root-MuiInput-root"
    >
      <select
        aria-invalid="false"
        class="MuiNativeSelect-select MuiNativeSelect-outlined MuiInputBase-input MuiInput-input css-19ygod6-MuiNativeSelect-select-MuiInputBase-input-MuiInput-input"
        name="select option"
      >
        <option
          value="option1"
        >
          Option 1
        </option>
        <option
          selected=""
          value="option2"
        >
          Option 2
        </option>
        <option
          value="option3"
        >
          Option 3
        </option>
      </select>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiNativeSelect-icon MuiNativeSelect-iconOutlined css-1g12qau-MuiSvgIcon-root-MuiNativeSelect-icon"
        data-testid="ArrowDropDownIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`Selector Component > should match snapshot 1`] = `
<div>
  <div
    class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-standard MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-standard css-1c2i806-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="true"
    >
      Select Option
    </label>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl MuiNativeSelect-root css-v4u5dn-MuiInputBase-root-MuiInput-root"
    >
      <select
        aria-invalid="false"
        class="MuiNativeSelect-select MuiNativeSelect-outlined MuiInputBase-input MuiInput-input css-19ygod6-MuiNativeSelect-select-MuiInputBase-input-MuiInput-input"
        name="select option"
      >
        <option
          value="option1"
        >
          Option 1
        </option>
        <option
          selected=""
          value="option2"
        >
          Option 2
        </option>
        <option
          value="option3"
        >
          Option 3
        </option>
      </select>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiNativeSelect-icon MuiNativeSelect-iconOutlined css-1g12qau-MuiSvgIcon-root-MuiNativeSelect-icon"
        data-testid="ArrowDropDownIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
    </div>
  </div>
</div>
`;
